### PR TITLE
Update readme: new feature set

### PR DESCRIPTION
**Pull request by ThHareau**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,16 +133,31 @@ If you still get lock timeout while adding / removing indexes, it might be for o
 - Long-running queries are active on the table. To create / remove an index, PG needs to wait for the queries that are actually running to finish before starting the index creation / removal. The blocking activity logger might help you to pinpoint the culprit queries.
 - A vacuum / autovacuum is running on the table, holding a ShareUpdateExclusiveLock, you are most likely out of luck for the current migration, but you may try to [optimize your autovacuums settings](https://www.percona.com/blog/2018/08/10/tuning-autovacuum-in-postgresql-and-autovacuum-internals/).
 
+</details>
+
+<details><summary>safe <code>add_foreign_key</code> (and <code>add_reference</code>)</summary>
 
-### Retry after lock timeout
+Adding a foreign key requires by default a `SHARE ROW EXCLUSIVE` lock, which conflicts with insert, update and delete statements for both referencing and referenced tables. In other words, **it is impossible to write in the tables** while the migration is running.
```
Comment (rchoquet) [inline README.md:140]:
```suggestion
Adding a foreign key requires a `SHARE ROW EXCLUSIVE` lock, which conflicts with insert, update and delete statements for both referencing and referenced tables. In other words, **it is impossible to write in the tables** while the migration is running.
```

File: README.md [hunk 2]
```diff
@@ -125,16 +133,31 @@ If you still get lock timeout while adding / removing indexes, it might be for o
...
+
+Adding the constraint is rather fast, the major part of the migration is spent on validating the new constraint. Thus safe-pg-migrations ensures that adding a foreign key is safe by splitting the addition of the foreign key in two parts:
```
Comment (rchoquet) [inline README.md:142]:
```suggestion
Adding the constraint itself is rather fast, the major part of the time is spent on validating the new constraint. Thus safe-pg-migrations ensures that adding a foreign key holds blocking locks for the least amount of time by splitting the foreign key creation in two steps: 
```

File: README.md [hunk 3]
```diff
@@ -125,16 +133,31 @@ If you still get lock timeout while adding / removing indexes, it might be for o
...
+</details>
+
+<details><summary>safe <code>add_foreign_key</code> (and <code>add_reference</code>)</summary>
 
-### Retry after lock timeout
+Adding a foreign key requires by default a `SHARE ROW EXCLUSIVE` lock, which conflicts with insert, update and delete statements for both referencing and referenced tables. In other words, **it is impossible to write in the tables** while the migration is running.
+
+Adding the constraint is rather fast, the major part of the migration is spent on validating the new constraint. Thus safe-pg-migrations ensures that adding a foreign key is safe by splitting the addition of the foreign key in two parts: 
+
+1. adding the constraint *without validation*;
```
Comment (rchoquet) [inline README.md:144]:
```suggestion
1. adding the constraint *without validation*, will not validate existing rows;
```

File: README.md [hunk 4]
```diff
@@ -125,16 +133,31 @@ If you still get lock timeout while adding / removing indexes, it might be for o
...
+2. validating the constraint.
```
Comment (rchoquet) [inline README.md:145]:
```suggestion
2. validating the constraint, will validate existing rows in the table, without blocking read or write on the table
```

File: README.md [hunk 5]
```diff
@@ -125,16 +133,31 @@ If you still get lock timeout while adding / removing indexes, it might be for o
...
 
-### Retry after lock timeout
+Adding a foreign key requires by default a `SHARE ROW EXCLUSIVE` lock, which conflicts with insert, update and delete statements for both referencing and referenced tables. In other words, **it is impossible to write in the tables** while the migration is running.
+
+Adding the constraint is rather fast, the major part of the migration is spent on validating the new constraint. Thus safe-pg-migrations ensures that adding a foreign key is safe by splitting the addition of the foreign key in two parts: 
+
+1. adding the constraint *without validation*;
+2. validating the constraint.  
+
+Indeed the validation of a constraint does not require any lock preventing reading or writing on the tables.
```
Comment (rchoquet) [inline README.md:147]: I'd remove this one then

Comment (ThHareau) [reply README.md:147]:
```suggestion
```

**Review by rchoquet — APPROVED** [review]

**Pull request merged**